dynmsg/src/msg_parser_c.cpp

```diff
@@ -335,109 +409,128 @@ void write_member_nested(const YAML::Node& yaml, uint8_t* buffer, const MemberIn
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_INT16: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_INT16:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_INT16>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_UINT32: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_UINT32:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_UINT32>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_INT32: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_INT32:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_INT32>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_UINT64: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_UINT64:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_UINT64>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_INT64: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_INT64:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_INT64>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_STRING: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_STRING:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_STRING>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_WSTRING: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_WSTRING:
         write_member<rosidl_typesupport_introspection_c__ROS_TYPE_WSTRING>(root, buffer, member);
         break;
-      }
-      case rosidl_typesupport_introspection_c__ROS_TYPE_MESSAGE: {
+      case rosidl_typesupport_introspection_c__ROS_TYPE_MESSAGE:
         write_member_nested(root, buffer, member);
         break;
-      }
-      default: {
+      default:
         throw std::runtime_error("unknown type");
-      }
     }
   }
 }
 
+}  // namespace impl
 
-RosMessage yaml_to_rosmsg(
-  const InterfaceTypeName &interface_type,
-  const std::string& yaml_str
-) {
+RosMessage yaml_to_rosmsg_typeinfo(
+  const TypeInfo * type_info,
+  const std::string & yaml_str,
+  rcutils_allocator_t * allocator)
+{
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  if (!allocator) {
+    allocator = &default_allocator;
+  }
   // Parse the YAML representation to an in-memory representation
   YAML::Node root = YAML::Load(yaml_str);
   RosMessage ros_msg;
   // Load the introspection information and allocate space for the ROS message's binary
   // representation
-  ros_message_init(interface_type, &ros_msg);
+  if (0 != dynmsg::c::ros_message_init_typeinfo(type_info, &ros_msg, allocator)) {
```

Comment:
Use the enum rather than 0?

Reply:
Ended up `typedef`ing a `dynmsg_ret_t` based on `rcutils_ret_t` with `DYNMSG_RET_OK` & `DYNMSG_RET_ERROR` in 4210cf005dcee897ce599d182ffa05441a4c4afb, since `dynmsg` shouldn't depend on `rcl`.